core/include/drivers/rstctrl.h

```diff
@@ -141,30 +141,42 @@ static inline bool rstctrl_ops_is_valid(const struct rstctrl_ops *ops)
 	return ops && ops->assert_level && ops->deassert_level;
 }
 
+#ifdef CFG_DT
 /**
- * rstctrl_dt_get_by_idx - Get a reset controller at a specific index in
+ * rstctrl_dt_get_by_index - Get a reset controller at a specific index in
  * 'resets' property
  *
  * @fdt: Device tree to work on
  * @nodeoffset: Node offset of the subnode containing a 'resets' property
  * @index: Reset controller index in 'resets' property
- * @res: Output result code of the operation:
- *	TEE_SUCCESS in case of success
- *	TEE_ERROR_DEFER_DRIVER_INIT if reset controller is not initialized
- *	Any TEE_Result compliant code in case of error.
+ * @rstctrl: Output reset controller reference upon success
  *
- * Returns a struct rstctrl pointer matching the reset controller at index
- * @index in resets property or NULL if no reset controller match the given
- * index in which case @res provides the error code.
+ * Return TEE_SUCCESS in case of success
+ * Return TEE_ERROR_DEFER_DRIVER_INIT if reset controller is not initialized
+ * Return a TEE_Result compliant code in case of error
  */
-static inline struct rstctrl *rstctrl_dt_get_by_idx(const void *fdt,
-						    int nodeoffset,
-						    unsigned int idx,
-						    TEE_Result *res)
+static inline TEE_Result rstctrl_dt_get_by_index(const void *fdt,
```

Comment:
Mindor: How about keeping this name as `rstctrl_dt_get_by_idx()` instead? Same for the idx parameter.

Reply:
It is rather for consistency with the change made in clk_dt_get_by*() that I change here `idx` to plain `index`.